types/serverless-step-functions/types/state.d.ts

```diff
@@ -0,0 +1,31 @@
+import { Choice } from './choice';
+import { Fail } from './fail';
+import { Map } from './map';
+import { Parallel } from './parallel';
+import { Pass } from './pass';
+import { Succeed } from './succeed';
+import { Task } from './task';
+import { Wait } from './wait';
+
+export type JsonPrimitive = string | number | boolean | null;
+export interface JsonObject {
+    [member: string]: JsonValue;
+}
+export interface JsonArray extends Array<JsonValue> {}
+export type JsonValue = JsonPrimitive | JsonObject | JsonArray;
+
+// https://docs.aws.amazon.com/step-functions/latest/dg/amazon-states-language-intrinsic-functions.html
+export type IntrinsicFunction = string;
+export type Resource = string | { 'Fn::GetAtt': [string, 'Arn'] } | { 'Fn::Join': [string, Resource[]] };
+
+export type JsonPath = string; // constraint: starts with $.
```

Comment:
There are two kinds of [paths](https://states-language.net/#path) in ASL. A `Path` and a `ReferencePath`. Both are effectively `string` to Typescript so there's no difference at compile time.

It could be useful from a tooling perspective to model them separately. For example, the schemas in asl-validator use separate names for these to enable validation of the expressions.

Also note that both can start with `$$.` when pulling data from the context